File: pet-management-platform/frontend/src/api/customer.ts
import request from '@/utils/request'

// 客户接口类型定义
export interface Customer {
  id?: number
  merchantId?: number
  name: string
  phone: string
  userId?: number
  gender?: string
  address?: string
  notes?: string
  totalOrders?: number
  totalAmount?: number
  lastVisitAt?: string
  status?: number
  createdAt?: string
  updatedAt?: string
}

// 获取客户列表
export function getCustomers() {
  return request({
    url: '/api/customers',
    method: 'get'
  }) as Promise<Customer[]>
}

// 搜索客户
export function searchCustomers(keyword: string) {
  return request({
    url: '/api/customers/search',
    method: 'get',
    params: { keyword }
  }) as Promise<Customer[]>
}

// 获取客户详情
export function getCustomerById(id: number) {
  return request({
    url: `/api/customers/${id}`,
    method: 'get'
  }) as Promise<Customer>
}

// 创建客户
export function createCustomer(data: Customer) {
  return request({
    url: '/api/customers',
    method: 'post',
    data
  }) as Promise<Customer>
}

// 更新客户
export function updateCustomer(id: number, data: Customer) {
  return request({
    url: `/api/customers/${id}`,
    method: 'put',
    data
  }) as Promise<Customer>
}

// 删除客户
export function deleteCustomer(id: number) {
  return request({
    url: `/api/customers/${id}`,
    method: 'delete'
  })
}

// 获取客户的宠物列表
export function getCustomerPets(customerId: number) {
  return request({
    url: `/api/user/pets/customer/${customerId}`,
    method: 'get'
  })
}

File: pet-management-platform/backend/pet-management-backend/src/main/java/com/petmanagement/petmanagementbackend/models/Customer.java
package com.petmanagement.petmanagementbackend.models;

import jakarta.persistence.*;
import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;
import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "customers")
@Data
@NoArgsConstructor
@AllArgsConstructor
public class Customer {
    
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;
    
    @Column(name = "merchant_id", nullable = false)
    private Long merchantId;
    
    @Column(name = "name", nullable = false, length = 50)
    private String name;
    
    @Column(name = "phone", nullable = false, length = 20)
    private String phone;
    
    @Column(name = "user_id")
    private Long userId;  // 关联系统用户（客户注册后绑定）
    
    @Column(name = "gender", length = 10)
    private String gender;
    
    @Column(name = "address")
    private String address;
    
    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;
    
    @Column(name = "total_orders")
    private Integer totalOrders = 0;
    
    @Column(name = "total_amount", precision = 10, scale = 2)
    private BigDecimal totalAmount = BigDecimal.ZERO;
    
    @Column(name = "last_visit_at")
    private LocalDateTime lastVisitAt;
    
    @Column(name = "status")
    private Integer status = 1;
    
    @CreationTimestamp
    @Column(name = "created_at", nullable = false, updatable = false)
    private LocalDateTime createdAt;
    
    @UpdateTimestamp
    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;
}